python/pypto/language/parser/ast_parser.py

```
@@ -171,6 +171,82 @@ def _fold_const_slice_extent(upper: object, lower: object) -> int | None:
+    def visit_BinOp(self, node: ast.BinOp) -> ast.expr:
+        node = self.generic_visit(node)
+        if isinstance(node.left, ast.Constant) and isinstance(node.right, ast.Constant):
+            try:
+                result = eval(
+                    compile(
+                        ast.fix_missing_locations(ast.Expression(body=node)),
+                        "<ct-fold>",
+                        "eval",
+                    ),
+                    {},
+                )
```

Ruff (0.15.18)

[error] Use of possibly insecure function; consider using `ast.literal_eval`

(S307)